### PR TITLE
bug 1269104 - Refactor compare_revisions tests, add translation tests

### PR DESCRIPTION
**Pull request by jwhitlock**
Convert existing ``CompareRevisionTests`` to ``py.test``, and split them between view tests (``test_view_revision.py``) and template tests (``test_templates.py``).  Add regression tests for using the view to compare a translation revision to the English source, as requested in PR #4304.

### CODE REVIEW

File: kuma/wiki/tests/test_views_revision.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+@pytest.mark.parametrize("id1,id2",
+                         [('1e309', '1e309'),
+                          ('', 'invalid'),
+                          ('invalid', ''),
+                          ])
+def test_compare_revisions_invalid_ids(root_doc, client, id1, id2):
+    """Comparing badly-formed revision parameters return 404, not error."""
+    url = urlparams(reverse('wiki.compare_revisions', args=[root_doc.slug],
+                            locale=root_doc.locale),
+                    **{'from': id1, 'to': id2})
```
Comment (escattone) [inline kuma/wiki/tests/test_views_revision.py]: I'd prefer `from=id1, to=id2`, but feel free to ignore.

File: kuma/wiki/tests/test_views_revision.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+def test_compare_revisions_wrong_document(edit_revision, client):
+    """If the revision is for the wrong document, a 404 is returned."""
+    doc = edit_revision.document
+    first_revision = doc.revisions.first()
+    other_doc = Document.objects.create(locale='en-US', slug='Other',
+                                        title='Other Document')
+    url = urlparams(reverse('wiki.compare_revisions', args=[other_doc.slug],
+                            locale=other_doc.locale),
+                    **{'from': first_revision.id, 'to': edit_revision.id})
```
Comment (escattone) [inline kuma/wiki/tests/test_views_revision.py]: I'd prefer `from=first_revision.id, to=edit_revision.id`, but feel free to ignore.

File: kuma/wiki/tests/test_views_revision.py (hunk 1 above)
Comment (jwhitlock) [inline kuma/wiki/tests/test_views_revision.py]: I did too, but ``from`` is a reserved keyword, and can't be used that way.

Comment (escattone) [inline kuma/wiki/tests/test_views_revision.py]: Oh, of course, never mind! 👍

**Pull request merged**